offlineimap/imaputil.py

```diff
@@ -370,3 +372,83 @@ def demodify(m):
         return ret.decode('utf-7').encode('utf-8')
     except (UnicodeDecodeError, UnicodeEncodeError):
         return name
+
+# Functionality to convert folder names encoded in IMAP_utf_7 to utf_8.
+# This is achieved by defining 'imap4_utf_7' as a proper encoding scheme.
+# Original code by (https://www.blogger.com/profile/16648963337079496096), from
```

Comment:
Somehow this comment got lost during reshaping of the branch:
http://piao-tech.blogspot.com/2010/03/get-offlineimap-working-with-non-ascii.html?showComment=1316041409339#c669880170006851138
indicates to me that the original author of this code expects it to be incorporated into offlineIMAP and therefore implicitly agrees to put it under its license.

Reply:
Add this credit to the commit message instead.

Reply:
This would require rebasing and thus changing the existing pushed commits. What I can do is adding *another* commit which removes the credit from the file and adds URL, reference to the author and mention of the license question in the commit message.
Should I do that @nicolas33? That's not perfectly clean either.

Reply:
Find the ancestor of your first commit and rebase on top of this. It's likely 1ce596d7135e.

You might like to tag this ancestor to have a pointer: `git tag utf8-ancestor 1ce596d7135e`.
This way it's possible to rebase as many times as needed to improve the patches (or squash already existing patches).

Reply:
The correct ancestor of this branch is v7.1.2. So, all the rebases can be done against tag v7.1.2.

Reply:
Hm.
My problem is not how to rebase. *If* I'm going to rebase I can as well do that on top of `next`. The problem is/was the question whether rebasing will destroy the review history - and if so if this is a problem.

Reply:
An compromise would be to finish the review until you are happy with it, and *then* I can rebase the changes in a new branch and push that as an alternative PR. So we'll have a clean history and you still have the original branch as a reference if anything should be unclear.